src/v/cluster/members_backend.cc

```diff
@@ -739,23 +741,24 @@ ss::future<> members_backend::reconcile() {
 }
 
 bool members_backend::should_stop_rebalancing_update(
-  const members_backend::update_meta& meta) const {
+  double current_error, const members_backend::update_meta& meta) const {
     // do not finish decommissioning and recommissioning updates as they have
     // strict stop conditions
-    auto error = calculate_unevenness_error();
-    vlog(
-      clusterlog.info,
-      "balance unevenness error - current: {}, previous: {}",
-      error,
-      _last_unevenness_error);
+
     if (
       meta.update
       && meta.update->type != members_manager::node_update_type::added) {
         return false;
     }
-    auto const stop_condition_improvement = 0.05;
-    // if improvement is negative, stop
-    auto improvement = std::max<double>(_last_unevenness_error - error, 0.0);
+    static auto const stop_condition_improvement = 0.05;
+
+    auto improvement = _last_unevenness_error - current_error;
```

Comment:
why isn't clamp to 0 needed now?

Reply:
negative is always smaller than `stop_condition`